### DISCUSSION TITLE
添加额外页面模块写在哪里？

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [vuepress/core#702] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/77786952/155082173-a61b3cc5-850a-42df-a91e-08d05db37acb.png)
这些代码是写在哪个文件中

**Comment by Linya-IronMan** [comment]
@meteorlxy  
1. 请问添加额外页面的时候，上面的这些代码要写在哪里？
2. 要写一个完全自定义的页面，就如平时开发页面一般，应该如何修改路由？

**Comment by Mister-Hope** [comment, marked as answer]
1. 自己建一个插件写在插件 api里。

```ts
// .vuepress/config.js

export default defineUserConfig({
  plugins: [
    {
      // 在这里使用 onIntialized
    }
  ]
});
```

2.
- 你可以在 clientAppEnhanceFiles 里拿到 router 对象以指定组件注册路由(当然这会失去 SSR)。（它更简单）
- 或者自行扩展主题注册一个新的布局，并在插件 API 里 `createPage` 并在 Front Matter 指定 `layout`(有SSR)。

**Reply by Linya-IronMan** [reply]
我要新增的是一个完全自定义的首页。参照上面的建议，通过 router 对象注册了路由。
```javascript
export default defineClientAppEnhance(({ app, router, siteData }) => {
  app.use(ArcoVue);
  router.addRoute({
    path: "/",
    component: () => import("../components/Home.vue"),
  });
});
```
---

现在遇到了另外一个问题。我在打包之后使用nginx进行部署。
在访问 localhost:5210  根目录的时候，因为没有对应的 html 文件，vue根本没有执行，所以 返回的是 404。
在访问其中一个文档之后，vue 有执行，再回到 根目录的时候，才能够访问到我自定义的页面。

---

尝试过在docs目录下写一个空的 README.md 作为首页，让router能够执行。但是页面会闪现一个空文档的vuepress渲染样式，然后才是我自定义的页面。

**Reply by Mister-Hope** [reply]
original reply is slightly edited to have a clearly description.
